### PR TITLE
Fix Pipeline Job titles are partially handled

### PR DESCRIPTION
**Pull request by uidoyen**
Closes: #1242

## Description
#1242
It fixes "_When a Pipeline Scheduled Run (aka a Job) creates Triggered Runs it creates it with a rather unpleasant name -- a fully lowercase, concatenated together string of the name of the Job._"

### Multiple runs delete
Screenshot 2023-07-14 at 4 54 41 PM

### Single run delete
Screenshot 2023-07-14 at 6 07 58 PM

### Run details - including bottom drawer
Screenshot 2023-07-14 at 4 54 04 PM

## How Has This Been Tested?

- Create scheduled run
- Go to triggered tab once scheduled run triggered
- Select multiple or single triggered run to delete run(s), on the delete modal the formatted  name(s)  should appear
- Click on the triggered run to go to the run details, see the formatted name on the page title and the bottom drawer

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [x] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Review by andrewballantyne — COMMENTED** [review]
I need a chance to review this -- I'm not sure I follow this implementation.

/hold

Let's not get this in for today's release.

**Comment by uidoyen** [comment]
> I need a chance to review this -- I'm not sure I follow this implementation.
> 
> /hold
> 
> Let's not get this in for today's release.

@andrewballantyne I am sorry, I just realized I misunderstood the implementation., hence adding label to WIP.

**Review by andrewballantyne — COMMENTED** [review]
@uidoyen Okay, sorry about the delay -- I finally got time to look this up and do a proper deep dive back on my original (last min) implementation that has left a lot desiring (hence the issue).

Let me explain the background, the "what" I did, and the place I think we need to get to (the last one is definitely something you can help us define).

Background:
* When we originally did the mocks and designed this feature, we were under the impression "Scheduled" runs were just runs with extra information about "rerunning"
* When implementation got to the plate, we realized there is this thing called Jobs. These Jobs are actually the "Scheduled" runs -- which in turn are not actually runs, but "things" that create runs
* When we started development and almost all the way to the end of the cycles, we didn't have Jobs working from the backend -- so we didn't see the UX issue until very late in the cycle
* DS Pipelines resources have something called `resource_references` -- this is an array of connections
    * Almost all resources have at least one of these (Experiments might not 🤔)
    * Pipelines often have an experiment reference (not sure if "Default" gets auto added if not specified)
    * Immediately executed runs often have a pipeline and experiment reference (I'm not sure how pure this is, if execution happens elsewhere outside of our UI, it might do something else)
    * Scheduled runs have a **JOB** and experiment reference -- this is where it differs and gets into the problematic space
        * if your run does not have a Pipeline reference, how do we know what Pipeline it was executed against?
        * Essentially this is done by going through the Job; Scheduled Run => Job => Pipeline (this is done through the `resource_reference` on each resource)
* Essentially this is all to say, without proper effort, we ended up getting something like `testkdswp-15-2214404825` being created as a run as the Job executed
    * This looked out of place and I brought this up to UX in our final review just before slapping the release sticker on the code (this predates incubation)
    * This also had no "Pipeline" reference because the run only pointed out it was executed from a Job

The last min fix:
* Having now seen how it looked -- odd name, no pipeline -- we realized it was a notable drop in usefulness to the user to see it without us trying to do something about it
* What I did was I walked to the Job resource ([how I did it in code](https://github.com/opendatahub-io/odh-dashboard/blob/main/frontend/src/concepts/pipelines/content/tables/pipelineRun/useJobRelatedInformation.ts)) as I needed it and acted on that resource set of information.
    * This was designed as a lazy-load mechanism and heavily relies on the setState re-triggering the render so the function can provide valuable info... we may want to change this to a rendering of the value as a ReactNode instead of as pure data (to make the rendering the same)
    * Problem is, I was narrow with my solution -- I did Pipeline Table rendering, as that's what we saw immediately during UX review
    * Delete Modals, Searching, Details Pages, none of these got the same treatment and thus show the old name

The place we need to get to:
* A unified UI rendering experience
    * Immediately executed runs still exist, they need to still render the desired way
    * Scheduled runs will be intermixed, they need to render nicer in all places they are rendered
    * They are interchangeable -- in all places both scenarios are in play, it can either be a Job's run, or a pure run
* An easy code-level way of getting this information in a way that is useful / helpful (maybe we keep my hook, maybe we toss it for something better)
    * Consider we might want to use a Component to just render the data and it can figure out which state to render in
* Low churn per use-case, an implementation needs to support the places where we need to fix this
    * Global Runs => Triggered Tab => Table row rendering
    * Global Runs => Triggered Tab => Table filtering
    * Global Runs => Triggered Tab => Delete modal; for delete one (row kebab) & delete many (checkboxes + the toolbar kebab)
    * Pipeline Run Details Page (this should be linkable from all 3 locations<sup>A</sup>) -- the name in the title needs to
    * I believe even duplicate run might use the old name and not the display name we chose to render -- you'll have to look into this as well
* Whatever our solution is, we need to make sure we configure the data once and then relay that information purely on rendering or in filters

    
Notes: 
* <sup>A</sup> - There are 3 locations in Pipelines. The Project Details, the Global Pipelines, and the Global Run pages. These three locations support a lot of the same pages (this is most notable in breadcrumbs, but also are supported by different routes). This is not perfect and there are gaps in some of the implementations around this
* It is worth noting, a scheduled run does not directly associate to a Pipeline, and thus never shows up in the Pipeline views -- since the Run is part of a Job. This is out of scope for this issue, but it is worth noting.

---

I get this is a lot to take in and not great to do this as a PR review. Let us talk more if you want, but take the time to absorb this information and understand the problem case more. Sorry to have made the ticket vague -- I expected to be the one to do this so my thoughts didn't really need to be documented per se.

**Comment by andrewballantyne** [comment]
/hold

Upcoming design problem with moving to API driven functionality. I started [a slack thread](https://redhat-internal.slack.com/archives/C043A9D5DMW/p1690399392667029) (internal) with UX.

[I inquired from the Pipeline team](https://odh-io.slack.com/archives/C03DY3YT151/p1690400555047179) (public) about possibly improving the Run names from Jobs.

**Comment by uidoyen** [comment]
### Runs - Recurring and Immediate Triggered Run

Screenshot 2023-08-01 at 11 39 05 PM

### Recurring Triggered Run details

Screenshot 2023-08-01 at 11 43 35 PM

### Immediate Triggered Run details
Screenshot 2023-08-01 at 11 46 53 PM

### Delete single run

Screenshot 2023-08-01 at 11 49 18 PM

### Delete multiple runs

Screenshot 2023-08-01 at 11 50 21 PM

**Comment by uidoyen** [comment]
@kywalker-rh @kaedward @yannnz For now, I have used `immediate`  and `recurring`. I will replace it once we finalize the term. Would you mind having a look into the PR?

**Review by lucferbux — CHANGES_REQUESTED** [review]
I have two issues, a part from the suggested changes proposed here https://docs.google.com/document/d/1moQoJNWCYhg_c9DjcdriM62gP2IJWSj623J6cMsIQJw/edit?usp=sharing

File: frontend/src/concepts/pipelines/content/PipelineRunTypeLabel.tsx
Comment (uidoyen) [reply frontend/src/concepts/pipelines/content/PipelineRunTypeLabel.tsx:14]: @lucferbux nice catch!, when we delete a `scheduled run` it looses `JOB` and `PIPELINE_VERSION` references. I have handled the condition accordingly. Thanks!

**Comment by pnaik1** [comment]
- [x] Able to see recurring and on-off tag
- [x] Able to see the small description about the scheduled run
Screenshot from 2023-08-07 11-31-34
- [x] The description part in the bottom drawer is working fine
Screenshot from 2023-08-07 12-20-18
- [x] Deleting single scheduled run working fine
- [x] Deleting multiple runs working fine 
- [x] Deleting Scheduled run Job working fine with scheduled run in triggered tab still showing recurring

**Comment by pnaik1** [comment]
I dont know whether this is related to this particular issue , But when we duplicate the recurring run the pipeline is not prefilled in the duplicate form ... 
https://github.com/opendatahub-io/odh-dashboard/assets/99238909/be656d4d-7971-4e42-8f8d-1447028878f2

**Comment by lucferbux** [comment]
> I dont know whether this is related to this particular issue , But when we duplicate the recurring run the pipeline is not prefilled in the duplicate form ... https://github.com/opendatahub-io/odh-dashboard/assets/99238909/be656d4d-7971-4e42-8f8d-1447028878f2 @lucferbux

mmm i think we addressed that recently, hold on.

**Review by manaswinidas — COMMENTED** [review]
1. Able to see `Recurring` and `One-off` labels
2. Description is reflected properly in both immediate and recurring runs on their respective pages, including the tables.
3. Deleting single and multiple runs works fine.
4. One concern(this may not be related): Even if you `Disable` in the run should we still be seeing `On-going` in the `Scheduled` row? Shouldn't that either be `stopped` or `Disabled`?
Screenshot 2023-08-07 at 6 11 13 PM

**Comment by uidoyen** [comment]
> * Deleting Scheduled run Job working fine with scheduled run in triggered tab still showing recurring
>   Is there anything I missed?



> 1. Able to see `Recurring` and `One-off` labels
> 2. Description is reflected properly in both immediate and recurring runs on their respective pages, including the tables.
> 3. Deleting single and multiple runs works fine.
> 4. One concern(this may not be related): Even if you `Disable` the run should we still be seeing `On-going` in the `Scheduled` row? Shouldn't that either be `stopped` or `Disabled`?
> 
> Screenshot 2023-08-07 at 6 11 13 PM

@manaswinidas regarding your comment we already have status enabled/disabled button to indicate the status `On-going` is based on ScheduledState

Screenshot 2023-08-07 at 6 32 37 PM
Screenshot 2023-08-07 at 6 33 37 PM

**Review by lucferbux — CHANGES_REQUESTED** [review]
Functionality works great, great job here!
I've left a few rundown of reviews, but once that's fixed we can merge this.

File: frontend/src/concepts/pipelines/content/DeletePipelineCoreResourceModal.tsx
Comment (uidoyen) [reply frontend/src/concepts/pipelines/content/DeletePipelineCoreResourceModal.tsx:147]: @lucferbux I agree, Maybe we can handle this in some of the improvement PR's

**Comment by lucferbux** [comment]
/hold

wait until next release

File: frontend/src/concepts/pipelines/content/PipelineJobReferenceName.tsx
Comment (andrewballantyne) [inline frontend/src/concepts/pipelines/content/PipelineJobReferenceName.tsx:21]:
```suggestion
          Run {getPipelineJobExecutionCount(resource.name)} of {data?.name}
```
Is there a reason to not just. use JSX?

File: frontend/src/concepts/pipelines/content/PipelineRunTypeLabel.tsx
Comment (andrewballantyne) [inline frontend/src/concepts/pipelines/content/PipelineRunTypeLabel.tsx:25]: I think we need tooltips in here -- it'll help users understand what these mean and why they are there.

Tooltips inline as comments.
```
      {jobReference ? (
        // Created by a scheduled run
        <Label color="blue">{PipelineRunLabels.RECURRING}</Label>
      ) : !pipelineReference ? (
        // Created by a scheduled run that was deleted
        <Label color="blue">{PipelineRunLabels.RECURRING}</Label>
      ) : (
        // Run once immediately after creation
        <Label color="blue">{PipelineRunLabels.ONEOFF}</Label>
      )}
```

cc @yannnz @kaedward

File: frontend/src/concepts/pipelines/content/pipelinesDetails/PipelineDetailsTitle.tsx
Comment (andrewballantyne) [inline frontend/src/concepts/pipelines/content/pipelinesDetails/PipelineDetailsTitle.tsx:11]: Params are `camelCase` unless it's a component function prop -- of which that would not be `boolean`. Boolean values are also often a pseudo-question... 

Suggest:
```suggestion
  includeTypeLabel?: boolean;
```

File: frontend/src/concepts/pipelines/content/tables/pipelineRun/PipelineRunTableRowTitle.tsx
Comment (andrewballantyne) [inline frontend/src/concepts/pipelines/content/tables/pipelineRun/PipelineRunTableRowTitle.tsx:19]: Is there a reason this is a variable and not just inline?

File: frontend/src/concepts/pipelines/content/tables/pipelineRun/usePipelineRunFilter.ts
Comment (andrewballantyne) [inline frontend/src/concepts/pipelines/content/tables/pipelineRun/usePipelineRunFilter.ts:37]: 43 requests to the same endpoint:
image

This is a performance problem -- we need to centralize this information. Maybe we need a Context or something -- maybe adding it to the `usePipelineAPI` so it can reuse values is what we need to do here.

**Review by andrewballantyne — CHANGES_REQUESTED** [review]
Some more changes needed. Looking really good though. We need to handle the network thing before we can consider merging this.

I'm thinking we may also want to take this to incubation first -- this seems like a big enough footprint. Maybe merge this into f/pipeline-enhancements instead of main?

File: frontend/src/concepts/pipelines/content/DeletePipelineCoreResourceModal.tsx
Comment (andrewballantyne) [reply frontend/src/concepts/pipelines/content/DeletePipelineCoreResourceModal.tsx:147]: Moreover... why did you do this?

Screenshot 2023-08-25 at 10 54 38 AM

This seems excessive repeating -- and it's not the translated value...

File: frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunDetails.tsx
Comment (andrewballantyne) [inline frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunDetails.tsx:1]: Can you fix a bug while you're here...

The delete modal has a weird condition:
```
      <DeletePipelineCoreResourceModal
        type="triggered run"
        toDeleteResources={deleting && run ? [run] : []}
        onClose={() => {
          navigate(contextPath ?? `/pipelineRuns/${namespace}`);
        }}
      />
```
Can we just close the modal -- seems odd to cancel the delete and navigate. Must have been a copy/paste logic fault on my part.

File: frontend/src/concepts/pipelines/content/tables/utils.ts
Comment (andrewballantyne) [inline frontend/src/concepts/pipelines/content/tables/utils.ts:153]: Please write these on this -- there is a failure state here you're not watching.

Also type your response from this function.

Comment (andrewballantyne) [reply frontend/src/concepts/pipelines/content/tables/utils.ts:153]:
```
getPipelineJobExecutionCount('test')
undefined
getPipelineJobExecutionCount('test-5-something')
'5'
getPipelineJobExecutionCount('test-something')
'test'
```
There are other use-cases here -- I think you need to take a stance on what this is intended on doing. Because `test` is not a count.

File: frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunTabDetails.tsx
Comment (andrewballantyne) [inline frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunTabDetails.tsx:97]: Did we not want to use the PipelineRun name here too? 
image

May be a bit redundant from the title -- but it is the current convention.

cc @yannnz

Comment (uidoyen) [reply frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunTabDetails.tsx:102]: @yannnz thanks for confirmation, I have added it back.

Comment (yannnz) [reply frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunTabDetails.tsx:102]: Thanks! I am not sure which screenshot should I look at as the latest one, but the order of the info is Name, Description, then following Pipeline, Project. @uidoyen

**Comment by uidoyen** [comment]
@andrewballantyne after moving `useJobRelatedInformation` to the context It always returns `null` I am sure i have done something wrong in the context implementation here `frontend/src/concepts/pipelines/context/PipelinesContext.tsx`

Screenshot 2023-08-31 at 8 56 28 PM

File: frontend/src/concepts/pipelines/context/PipelinesContext.tsx
Comment (andrewballantyne) [inline frontend/src/concepts/pipelines/context/PipelinesContext.tsx:168]: Delete this -- thread the value from:
```
  const {
    hasCR,
    crInitializing,
    serverTimedOut,
    apiState,
    namespace,
    project,
    refreshAPIState: refreshAllAPI,
  } = React.useContext(PipelinesContext);
```

So once you delete it -- the value you import will take its place:
```
const {
  ...
  getJobInformation,
} = React.useContext(PipelinesContext);
```
And this should make it work.

**Review by andrewballantyne — CHANGES_REQUESTED** [review]
> @andrewballantyne after moving `useJobRelatedInformation` to the context It always returns `null` I am sure i have done something wrong in the context implementation here `frontend/src/concepts/pipelines/context/PipelinesContext.tsx`
> 
> Screenshot 2023-08-31 at 8 56 28 PM

@uidoyen This should help solve your issue I imagine

File: frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunTabDetails.tsx
Comment (uidoyen) [reply frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunTabDetails.tsx:100]: @yannnz here is the latest screenshot.

Screenshot 2023-08-31 at 10 16 14 PM

Comment (yannnz) [reply frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunTabDetails.tsx:104]: Sorry I missed this message somehow. I just realized why don't we have Pipeline info in the Recurring run's detail? Is there a technical reason? @uidoyen

File: frontend/src/concepts/pipelines/content/createRun/useRunFormData.ts
Comment (andrewballantyne) [inline frontend/src/concepts/pipelines/content/createRun/useRunFormData.ts:80]: I'm still working on this -- we should re-use existing details and only apply our description if somehow the resource doesn't have one. We should not touch the name -- that can be picked up from the earlier flows.

Ashique and I talked about this -- I wasn't able to work out the TS logic in time. We can swing back.

File: frontend/src/concepts/pipelines/content/createRun/useRunFormData.ts
Comment (andrewballantyne) [inline frontend/src/concepts/pipelines/content/createRun/useRunFormData.ts:203]: I think this logic is dealt with in other spots -- we may want to look for those and make this a unified utility.

**Review by andrewballantyne — APPROVED** [review]
This PR has moved on from the original description (eg outdated images) -- but it also has lingered for a while due to me prioritizing other tasks.

There is a lot of good work here -- I will log a follow up to catch the things I think need to still be done. But let us get the last 2 months of work in as a foundation.

Thanks for working with me to get it to a place that I am happy with @uidoyen -- I'll log an issue and point at this review for the items.

File: frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunTabDetails.tsx
Comment (andrewballantyne) [reply frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunTabDetails.tsx:104]: Fixed -- it is in the Job not in the Runs created by Jobs like normal Runs.

**Pull request merged**
